email-common/src/main/java/io/commercetools/sunrise/email/MessageEditor.java

```diff
@@ -0,0 +1,21 @@
+package io.commercetools.sunrise.email;
+
+import javax.annotation.Nonnull;
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+
+/**
+ * A functional interface for modifying a message without needing to handle checked exceptions like the
+ * {@link MessagingException}s thrown by {@link MimeMessage}.
+ */
+@FunctionalInterface
+public interface MessageEditor {
```

Comment:
I would go for plain examples, since the components of this module are quite small to forget to update the documentation, and shouldn't be changing much. So if we change the Interface itself, we only have to take care of two classes. For any implementation, it's only the class itself.